services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -493,13 +493,54 @@ specDeleteUser = do
                 !!! const 405 === statusCode
 
     describe "DELETE /Users/:id" $ do
+        it "should be delete successfully and be idempotent" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let uid = scimUserId storedUser
+
+            spar <- view teSpar
+            deleteUser_ (Just tok) (Just uid) spar
+                !!! const 204 === statusCode
+            deleteUser_ (Just tok) (Just uid) spar
+                !!! const 204 === statusCode
+
+
+        it "should return 401 if we don't provide a token" $ do
```

Comment:
This test is already present in hscim.

Reply:
> FUTUREWORK: hscim has the the following test.  we should probably go through all
> `delete` tests and see if they can move to hscim or are already included there.

Why add the `FUTUREWORK` if we can just not add the test in the first place?

Reply:
@fisx You added this note; thoughts?

I'm personally reluctant to have cross-cutting knowledge like this reduce our test coverage. E.g. if we want spar to return 401 when provided no token, then Spar should have a test for that; eventually hscim may change or we may stop using it entirely; but this is an integration test so I think it should assert that this is the behaviour here. If hscim also has a similar test that's great; but I don't think it makes this one redundant.

Reply:
> I'm personally reluctant to have cross-cutting knowledge like this reduce our test coverage.

Sure, but then if you want to be consistent you should have this test ("are we doing authorization correctly?") for literally every endpoint. And then somebody else can take this further and also say that we should test that `servant-server` is correct. Etc, etc.

It's nice to draw a line in the sand because otherwise people can not make PRs without being uncertain, and uncertainty is a productivity killer. (I already experience this whenever I have to decide how to name a function, or in which module that function should live.)

If you think authorization is important enough that it should be double-tested, then at least it should be a separate "is our SCIM library sane?" kind of test in a new test branch.

Alternatively, if you think authorization should be tested for every endpoint, I'd like to know where your line in the sand is.

Reply:
Other than this, the PR looks good to me.

Reply:
In my history my standard has been YES authorization should be tested for every endpoint. This is also a good habit to get into, because then I simply write these tests for every endpoint I write. In most other services (galley, brig, etc.) We don't have centralized auth checks and so we SHOULD be writing tests for every endpoint. If you insist I remove this additional test coverage then I can; but I'm going to continue writing auth tests for new endpoints in other services, especially since it's not always clear exactly which endpoints are backed by SCIM and which aren't; and also we need to know which endpoints are backed by SCIM and WILL NEVER be changed to anything else; because if we change then suddenly we have ZERO test coverage on all those endpoints (for auth) 😋